### ISSUE TITLE
[native--image] Kerberos: Generated image is not respecting sun.security.jgss.native

### ISSUE THREAD

**Issue by aanavaneeth** [oracle/graal#8674]
As per https://docs.oracle.com/en/java/javase/11/security/accessing-native-gss-api.html#GUID-88A42A9C-AC8B-426E-A8A7-B30518C2162A, when `sun.security.jgss.native` flag is set to `true`, `GSSManager.getInstance()` should ideally give a GSSManager implementation using native gss implementation. However, that is not happening.

**Comment by aanavaneeth** [comment]
Is there any issue with features leveraging jvm arguments in general? I am facing Kerberos related issue which depends on a jvm argument. https://github.com/oracle/graal/issues/8674

**Comment by aanavaneeth** [comment]
Hi @fernando-valdez ! Any update on this?

**Comment by yogeshkumar-1234** [comment]
Hi @fernando-valdez ,
I am facing similar issue .
Kindly help

**Comment by yogeshkumargeek** [comment]
Hi @oubidar-Abderrahim , 
Can you please provide me an update here ? 
This is a blocker for us also

**Comment by gingk1212** [comment]
I found this issue interesting, so I looked into it during my free time. Regarding the sample code in this issue, the result of the `findLoadedClass` method remains `null`. However, for the sample code in #4700, I discovered an option that prevents the error, and I would like to share it:

```
$ javac Sample.java
$ java \
    -agentlib:native-image-agent=config-output-dir=META-INF/native-image \
    -Dsun.security.jgss.native=true \
    Sample
$ native-image \
    --no-fallback \
    --initialize-at-run-time=sun.security.jgss.wrapper.Krb5Util,sun.security.jgss.wrapper.SunNativeProvider \
    Sample
$ ./sample -Dsun.security.jgss.native=true
```

With these commands, the error does not occur, and the program finishes successfully. Since the `GSSManagerImpl` class that uses the `sun.security.jgss.native` property is initialized at runtime, I believe the option must be specified at runtime rather than at build time.

I don’t fully understand all the details, but I hope this information is helpful.

Environment:

```
$ java -version
java version "23.0.1" 2024-10-15
Java(TM) SE Runtime Environment Oracle GraalVM 23.0.1+11.1 (build 23.0.1+11-jvmci-b01)
Java HotSpot(TM) 64-Bit Server VM Oracle GraalVM 23.0.1+11.1 (build 23.0.1+11-jvmci-b01, mixed mode, sharing)
```

**Comment by jovanstevanovic** [comment]
@gingk1212 Yes, you were right, three things were missing in this reproducer. The first was the agent run (the first command you mentioned), and the second was the system property. The system property should be specified at run time in the way you described. The third one that I had to add was `--initialize-at-run-time=sun.security.jgss.wrapper` on command line.

I ran it locally on master and confirmed that with these changes, the issue is resolved.
